redis/src/sentinel.rs

```diff
@@ -302,19 +302,24 @@ fn determine_master_from_role_or_info_replication(
     let mut conn = client.get_connection()?;
 
     //Once the client discovered the address of the master instance, it should attempt a connection with the master, and call the ROLE command in order to verify the role of the instance is actually a master.
-    let role = check_role(&mut conn);
-    if role.is_ok_and(|x| matches!(x, Role::Primary { .. })) {
-        return Ok(true);
+    let role_result = check_role(&mut conn);
+    if let Ok(role) = role_result {
+        return Ok(matches!(role, Role::Primary { .. }));
     }
 
     //If the ROLE commands is not available (it was introduced in Redis 2.8.12), a client may resort to the INFO replication command parsing the role: field of the output.
-    let role = check_info_replication(&mut conn);
-    if role.is_ok_and(|x| x == "master") {
-        return Ok(true);
+    let fallback_role_result = check_info_replication(&mut conn);
+    if let Ok(role) = fallback_role_result {
+        return Ok(role == "master");
     }
 
-    //TODO: Maybe there should be some kind of error message if both role checks fail due to ACL permissions?
-    Ok(false)
+    let role_err = role_result.unwrap_err();
+    // unknown commands are expressed as response errors
+    if !matches!(role_err.kind(), ErrorKind::ResponseError) {
+        return Err(role_err);
+    }
+
+    Err(fallback_role_result.unwrap_err())
```

Comment:
the end of the function is repeated twice (or three times, with `async_determine_slave_from_role_or_info_replication`). Can you please move it to a function, for DRY reasons?

Reply:
> Sure, but why is there no blocking version of that method?

I assume because nobody contributed it 🤷